### DISCUSSION TITLE
table with virtual columns

### DISCUSSION THREAD

**Discussion by lschierer** [adobe/spectrum-web-components#3737] [Ideas]
Right now it is possible to have a table with static columns and virtual data.  Has a version of the table with virtual columns been considered?  The use case I am pondering is a dataset that has more columns than can be usefully viewed at one time, and allowing the user to select which ones to display.  I tried changing the columns after the sp-table is rendered, but the added columns cause the table formatting to behave oddly.

**Comment by Westbrook** [comment]
This is not currently under consideration. Our virtualization tool does not support multi axis virtualization at this time. It would likely be a decent amount of work to add something like this, in that case. If you were interested in structuring a possible contribution, we’d be interested in seeing it.
